File: Array&String/Stack/remove_outermost_parentheses.py
class Solution:
    def removeOuterParentheses(self, s: str) -> str:
        ans, l, r, cur_idx, n = "", 0, 0, 0, len(s)
        if n == 1: return s

        for i, el in enumerate(s):

            if el == '(': l += 1
            else: r += 1

            if l == r:
                ans += s[cur_idx + 1 : i]
                cur_idx, l, r = i + 1, 0, 0

        return ans

File: Array&String/Stack/CPP/remove_outermost_parentheses.cpp
#include <string>
using namespace std;

class Solution {
public:
    string removeOuterParentheses(string s) {
        int l = 0, r = 0, idx = 0, n = s.size();
        string res = "";

        for (int i = 0; i < n; i++) {
            if (s[i] == '(') l++;
            else r++;

            if (l == r) {
                res += s.substr(idx + 1, (i - (idx + 1)));
                idx = i + 1;
                l = 0;
                r = 0;
            }
        }
        return res;
    }
};